scripts/scrub.js

```diff
@@ -87,19 +107,30 @@ async function run() {
     }
   }
 
-  console.log("Deleting symlinks from packages' node_modules...");
-  deleteSymlinks('apps');
-  deleteSymlinks('packages');
+  const failedPaths = [];
+
+  console.log("\nDeleting symlinks from packages' node_modules and rush temp files...");
+  const folders = [...getChildren('apps'), ...getChildren('packages'), 'scripts', 'common/temp'];
+  for (const folder of folders) {
+    deleteNodeModulesSymlinks(folder, failedPaths);
+  }
+  deleteIfSymlink(path.resolve('common/temp/pnpm-local'), failedPaths);
+
+  if (failedPaths.length) {
+    console.error('Deleting the following symlinks failed. Please delete these manually and try again.');
+    console.error(failedPaths.map(p => '  ' + p).join('\n'));
+    process.exit(1);
+  }
 
-  console.log('Deleting symlinks from rush temp files...');
-  deleteNodeModulesSymlinks('common/temp/node_modules');
-  deleteIfSymlink('common/temp/pnpm-local');
+  console.log('\nRunning "git clean -fdx" to remove all untracked files/folders (this may take awhile)...');
+  await execWithPipe('git clean -fdx');
+  // Run it again in case the first time didn't fully work (known to happen)
+  await execWithPipe('git clean -fdx');
```

Comment:
I'm willing to help you characterize this more. The problem is I think if this needs to be done, it also indicates the repo is in a bad state with deleted controlled files. If you want to make a 6.0 branch version of this script I can test it out.

Reply:
I removed the extra clean but added `git reset --hard` to help clean up if tracked files accidentally get deleted. (Also realized I forgot to clean up symlinks from `scripts/node_modules`, so some files were getting deleted because of that.) After this PR goes in I also plan to check it in to 6.0.

Reply:
I can't get the situation to repro now. I thought it may be because I had VSCode open or even something running with `npm start` but all will cause `git clean -fdx` to continue or block as expected. I'm not sure what the script would do, but running it while a webpack server is launched may be a good test, assuming you haven't done that already.

Reply:
I added code to make the script bail out (before git clean) if it fails to delete any symlinks, which I think would help handle that case.

Reply:
Or at least, it would help prevent accidental deletion of tracked files due to symlinks that failed to delete.